File: content/snippets/scala3/language_basics_nullable_optional_values.scala
object NullableOptionalValuesBasics:
  def parseAge(text: String): Option[Int] =
    if (text.isEmpty) None else Some(text.toInt)

  @main def main(): Unit =
    val age = parseAge("29")
    val missing = parseAge("")

    println(s"age present: ${age.isDefined}")
    println(s"missing default: ${missing.getOrElse(0)}")

File: content/snippets/python/language_basics_nullable_optional_values.py
#!/usr/bin/env python3
from typing import Optional


def parse_age(text: str) -> Optional[int]:
    if not text:
        return None
    return int(text)


def main() -> None:
    age = parse_age("29")
    missing = parse_age("")

    print("age present:", age is not None)
    print("missing default:", missing or 0)


if __name__ == "__main__":
    main()
